test/spec/features/snapping/BpmnCreateMoveSnappingSpec.js

```diff
@@ -715,6 +715,38 @@ describe('features/snapping - BpmnCreateMoveSnapping', function() {
 
   });
 
+  describe('dragging elements', function() {
+    var diagramXML = require('./BpmnCreateMoveSnapping.process.bpmn');
+
+    beforeEach(bootstrapModeler(diagramXML, {
+      modules: testModules
+    }));
+
+    it('should limit movement when attempting to drag outside the viewport', inject(function(elementRegistry, move, dragging, canvas) {

```

Comment:
I've reverted the change to the rules, and the test still passes. What we'd like to achieve is that the test is failing without the change you introduced: https://github.com/bpmn-io/bpmn-js/pull/2244/files#diff-6031dc6e8e3d4e38aef3c1dda35e6ff836a6b6be1263157300284d0191a6d5d4R931
Otherwise, the test does not verify the fix as it is passing without any changes to the library code.

Reply:
Would it be possible to view some documentation on how the inject function works? Trying to figure out how to simulate the viewport bc in the test it does move outside but it isn't perceived as a viewport so it doesn't snap back if that makes sense.

Reply:
@Adam-Thometz inject will execute the test within a previously bootstrapped bpmn-js instance, and allows you to access internal services.

Similar to this:

```javascript
const bpmnJS = new BpmnModeler();

const testFn = function(eventBus) { ... };

bpmnJS.invoke(testFn);
```